File: 0056-merge-intervals/0056-merge-intervals.java
class Solution {
    public int[][] merge(int[][] intervals) {
        
        Arrays.sort(intervals, (a, b) -> Integer.compare(a[0], b[0]));
        
        ArrayList<int[]> ans = new ArrayList<>();
        
        for (var pairs : intervals) {
            if (ans.isEmpty() || ans.get(ans.size() - 1)[1] < pairs[0]) {
                ans.add(pairs);
            } else {
                ans.get(ans.size() - 1)[1] = Math.max(ans.get(ans.size() - 1)[1], pairs[1]);
            }
        }
        
        return ans.toArray(new int[ans.size()][2]);
    }
}

File: 0056-merge-intervals/0056-merge-intervals.cpp
class Solution {
public:
    vector<vector<int>> merge(vector<vector<int>>& intervals) {
        std::sort(intervals.begin(), intervals.end());
        vector<vector<int>> ans;
        for(auto pairs : intervals){
            if(ans.empty() or ans.back()[1] < pairs[0]){
                ans.push_back(pairs);
            }else{
                ans.back()[1] = max(ans.back()[1], pairs[1]);
            }
        }
        return ans;
    }
};